KHOULOUDouel/myFlix-client
# Task 7

Title: Implement State-Based Routing and User Profile View

Description: This pull request introduces state-based routing to the myFlix-client application using React Router, along with a user profile view where users can view and update their information, and manage their favorite movies. The task has been divided into several parts, as outlined below:

Part 1: Implement State-Based Routing & Navigation
Setup Routing:

Created a new branch off the main branch.
Installed react-router and react-router-dom packages.
Refactored main-view.jsx to use BrowserRouter and added routes for various views.
Replaced buttons using onMovieClick and onBackClick with React Router links in MovieCard and MovieView.
Removed state management related to selectedMovie.
Navigation Bar:

Added a navigation bar at the top of the page.
Provided links to Login/Signup for unauthenticated users.
Provided links to Home (MainView), Profile, and Logout for authenticated users.
Changes:

Refactored main-view.jsx to use React Router.
Updated MovieCard and MovieView components.
Added navigation-bar.jsx component.
Commit Summary:

Installed React Router packages.
Implemented routing in main-view.jsx.
Replaced state-based navigation with React Router links.
Added navigation bar.
Part 2: Create a User Profile View
Profile View:

Created a new profile-view.jsx component.
Displayed user information obtained via the /users endpoint.
Allowed users to update their information (username, password, email, date of birth).
Allowed users to deregister.
Displayed user's favorite movies using MovieCard components.
Favorite Movies:

Added a "Favorite" button to MovieCard and MovieView components for logged-in users.
Enabled users to add or remove movies from their list of favorites.
Changes:

Added profile-view.jsx component.
Implemented functionality to display and update user information.
Implemented functionality to manage user's favorite movies.
Commit Summary:

Created profile view for users.
Displayed and updated user information.
Managed user's favorite movies.Additional Information:

Ensured that the backend server is running and accessible from the frontend.
Fixed various JSX structure issues.
Applied all best practices learned during the course.
How to Test:

Start the backend server.
Run the frontend application.
Navigate through different views using the navigation bar.
Verify user authentication and access to protected routes.
Test user profile functionalities including viewing and updating user information and managing favorite movies.
Task Breakdown:

Implemented state-based routing.
Added a navigation bar.
Created a user profile view.
Managed user favorite movies.
